File: typescript/examples/basic.ts
// node --loader tsx examples/basic.ts
//   PLUMR_API_KEY=plm_live_… npx tsx examples/basic.ts

import Plumr from "@plumr/sdk";

const plumr = new Plumr({
  apiKey: process.env.PLUMR_API_KEY ?? "",
});

const stream = plumr.run({
  input: process.argv.slice(2).join(" ") || "Write a haiku about Mars.",
});

for await (const event of stream) {
  if (event.type === "llm.delta") {
    process.stdout.write(event.text);
  } else if (event.type === "run.end") {
    process.stdout.write("\n");
    console.error(`done · ${event.durationMs}ms · ${event.status}`);
    if (event.error) console.error("error:", event.error);
  }
}

File: python/examples/basic.py
"""Minimal streaming example.

    PLUMR_API_KEY=plm_live_... python examples/basic.py "your prompt"
"""

import os
import sys

from plumr import Plumr


def main() -> None:
    api_key = os.environ.get("PLUMR_API_KEY")
    if not api_key:
        sys.exit("Set PLUMR_API_KEY first.")

    prompt = " ".join(sys.argv[1:]) or "Write a haiku about Mars."

    with Plumr(api_key=api_key) as client:
        for event in client.run(input=prompt):
            if event.type == "llm.delta":
                print(event.text, end="", flush=True)
            elif event.type == "run.end":
                print()
                print(
                    f"done · {event.durationMs}ms · {event.status}",
                    file=sys.stderr,
                )
                if event.error:
                    print(f"error: {event.error}", file=sys.stderr)


if __name__ == "__main__":
    main()
